src/osd/PrimaryLogPG.cc

```diff
@@ -1844,10 +1844,8 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
   ceph::ref_t<Session> session;
   if (can_backoff) {
     session = static_cast<Session*>(m->get_connection()->get_priv().get());
-    if (!session.get()) {
-      dout(10) << __func__ << " no session" << dendl;
-      return;
-    }
+    // do_request already check
+    ceph_assert(session.get());
```

Comment:
```
2020-04-13T03:11:57.821 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: /home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/cent
os8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-704-g2d90622/rpm/el8/BUILD/ceph-16.0.0-704-g2d90622/src/osd/PrimaryLogPG.cc: 1868: FAILED ceph_assert(session.get())
2020-04-13T03:11:57.822 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: ceph version 16.0.0-704-g2d90622 (2d906227446ef9114a7b0b01746561e259d5ae91) pacific (dev)
2020-04-13T03:11:57.822 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x158) [0x55ad4408dd68]
2020-04-13T03:11:57.822 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 2: (()+0x50bf82) [0x55ad4408df82]
2020-04-13T03:11:57.822 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 3: (PrimaryLogPG::do_op(boost::intrusive_ptr<OpRequest>&)+0x4444) [0x55ad442ddd54]
2020-04-13T03:11:57.823 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 4: (PrimaryLogPG::do_request(boost::intrusive_ptr<OpRequest>&, ThreadPool::TPHandle&)+0xd40) [0x55ad442ea380
]
2020-04-13T03:11:57.824 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 5: (OSD::dequeue_op(boost::intrusive_ptr<PG>, boost::intrusive_ptr<OpRequest>, ThreadPool::TPHandle&)+0x309)
 [0x55ad44171689]
2020-04-13T03:11:57.825 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 6: (ceph::osd::scheduler::PGOpItem::run(OSD*, OSDShard*, boost::intrusive_ptr<PG>&, ThreadPool::TPHandle&)+0
x68) [0x55ad443cc738]
2020-04-13T03:11:57.825 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 7: (OSD::ShardedOpWQ::_process(unsigned int, ceph::heartbeat_handle_d*)+0x12ef) [0x55ad4418e90f]
2020-04-13T03:11:57.826 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 8: (ShardedThreadPool::shardedthreadpool_worker(unsigned int)+0x5c4) [0x55ad447c58e4]
2020-04-13T03:11:57.826 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 9: (ShardedThreadPool::WorkThreadSharded::entry()+0x14) [0x55ad447c8544]
2020-04-13T03:11:57.826 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 10: (()+0x82de) [0x7fb7c47122de]
2020-04-13T03:11:57.826 INFO:ceph.osd.6.smithi014.stdout:Apr 13 03:11:57 smithi014 bash[72876]: 11: (clone()+0x43) [0x7fb7c34bc133]
```

http://pulpito.ceph.com/kchai-2020-04-13_01:52:52-rados-wip-kefu-testing-2020-04-13-0113-distro-basic-smithi/4948174/

Reply:
@tchaikov . I review the code. for get_priv/set_priv/clear_priv, they protected by lock. So i think we can't assume if in do_request, get_priv isn't null, in do_op get_priv must not-null.  So i'll close this PR.

Reply:
thanks for looking into the root cause, Jianpeng!